packages/ember-metal/lib/properties.js

```diff
@@ -38,6 +41,16 @@ export function DEFAULT_GETTER_FUNCTION(name) {
   };
 }
 
+export function INHERITING_GETTER_FUNCTION(name) {
+  function IGETTER_FUNCTION() {
+    var proto = Object.getPrototypeOf(this);
```

Comment:
It seems to me (just off the bat) that we need to do the same thing as was added in https://github.com/emberjs/ember.js/pull/12314 to lookup the getter through the entire prototype chain.  As it exists here, I believe that a getter from a parent class will not be found.

Specifically, I think we should use [lookupDescriptor](https://github.com/emberjs/ember.js/blob/bb5cc203c79da47af1ead291848c280c63ee6fe5/packages/ember-metal/lib/watch_key.js#L40-L53), and add some tests around this behavior like [these ones](https://github.com/emberjs/ember.js/blob/bb5cc203c79da47af1ead291848c280c63ee6fe5/packages/ember-metal/tests/accessors/mandatory_setters_test.js#L97-L153) only for getter.